### PR TITLE
integrate pennsieveClient to retrieve dataset and metadata information

### PR DESCRIPTION
**Pull request by edmore**
- adds calls to reference new discover clients in pennsieve-go
- passes variables from lambda to ECS task for processing

### CODE REVIEW

File: fargate/rehydrate/main.go [hunk 1]
```diff
@@ -1,7 +1,49 @@
...
+	log.Println(datasetByVersionReponse)
+
+	datasetMetadataByVersionReponse, err := pennsiveClient.Discover.GetDatasetMetadataByVersion(ctx, int32(datasetId), int32(versionId))
+	if err != nil {
+		log.Fatalf("error retrieving dataset by version")
+	}
+	log.Println(datasetMetadataByVersionReponse)
+}
+
+func getApiHost(env string) string {
```
Comment (edmore) [inline fargate/rehydrate/main.go:43]: not sure if there is a cleaner way to do this.

File: lambda/service/mocks/mocks.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+type MockECSTaskRunner struct{}
+
+func (r *MockECSTaskRunner) Run(ctx context.Context) error {
+
+	return nil
+}
+
+func NewMockECSTaskRunner() runner.Runner {
+	return &MockECSTaskRunner{}
+}
```
Comment (edmore) [inline lambda/service/mocks/mocks.go:18]: adds mocks for use later

File: fargate/rehydrate/main.go [hunk 3]
```diff
@@ -1,7 +1,49 @@
...
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/pennsieve/pennsieve-go/pkg/pennsieve"
+)
 
 func main() {
 	log.Println("Running rehydrate task")
+	ctx := context.Background()
```
Comment (edmore) [inline fargate/rehydrate/main.go:14]: there will be a follow-up MR to do the actual copy used the retrieved data.

**Review by muftring — APPROVED** [review]
lgtm - just a couple comments; pick them up in the next PR

File: fargate/rehydrate/main.go (hunk 1 above)
Comment (edmore) [reply fargate/rehydrate/main.go:43]: Noted.

File: fargate/rehydrate/main.go [hunk 4]
```diff
@@ -1,7 +1,49 @@
...
+	datasetId, err := strconv.Atoi(os.Getenv("DATASET_ID"))
+	if err != nil {
+		log.Fatalf("err converting datasetId to int")
+	}
+	versionId, err := strconv.Atoi(os.Getenv("DATASET_VERSION_ID"))
+	if err != nil {
+		log.Fatalf("err converting versionId to int")
+	}
+
+	pennsiveClient := pennsieve.NewClient(pennsieve.APIParams{ApiHost: getApiHost(os.Getenv("ENV"))})
```
[the remark this answers is not in the document]
Comment (edmore) [reply fargate/rehydrate/main.go]: updated, thanks.

**Pull request merged**